### DISCUSSION TITLE
Is there possibility of receiving layer change event from device?

### DISCUSSION THREAD

**Discussion by Legolek** [maartentamboer/X-Touch-Mini-FS2020#85] [Q&A] [answered]
Hello,
First of all amazing work. Keep using your tool and now decided to code some features especially for my use cases. I'm wondering if changing layer on device can be somehow received by software. I see your coding that it happens on first button or encoder usage after layer is changed so I assume it's not, as it would be much easier if message would be sent from device.

Is it device issue or protocol, do you maybe know?

Best Regards,
Michal

**Comment by maartentamboer** [comment, marked as answer]
Hi Michal, 

Unfortunately the X-touch does not send a midi message when you change a layer and that is indeed the reason for the workaround you see in the code. If you want to see for yourself undo this comment to view all the midi messages coming from the device: https://github.com/maartentamboer/X-Touch-Mini-FS2020/blob/main/midiconnection.py#L28

It is possible to change the layer from the PC by sending a `Program Change` message, maybe tie that to one of the regular buttons. For more information see https://mediadl.musictribe.com/media/PLM/data/docs/P0B3M/X-TOUCH%20MINI_QSG_WW.pdf 
Page 8 of the PDF (14, 15 from the manual)
